### PR TITLE
feat(design): add cli to build Design Token from several Design Token

### PR DESCRIPTION
**Pull request by kpanot**
Add cli to build Design Token from several Design Token file and check duplication

### CODE REVIEW

File: packages/@o3r/design/README.md [hunk 1]
```diff
@@ -67,6 +67,25 @@ The following configurations are available:
...
 
+## Command Line Interfaces
+
+### o3r-build-design-token
+
+The purpose of this CLI is to concatenate different Design Token files and check the potential duplication.
+Can be run with the following command:
+
+```shell
+npx -p @o3r/design o3r-build-design-token <...design-tokens> -o ./my-new-token-fils.json
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/README.md:78]:
```suggestion
npx -p @o3r/design o3r-build-design-token <...design-tokens> -o ./my-new-token-file.json
```

File: packages/@o3r/design/cli/build-design-token.cli.cts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    renderDesignTokenOptions.determineFileToUpdate = () => resolve(process.cwd(), output);
+  }
+
+  const tokens = (await Promise.all(
+    args._
+      .map((file) => isAbsolute(file) ? normalize(file) : resolve(process.cwd(), file))
+      .filter((file) => {
+        const res = existsSync(file);
+        if (!res) {
+          throw new Error(`The file ${file} does not exist, the process will stop`);
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/cli/build-design-token.cli.cts:29]:
```suggestion
          throw new Error(`The file ${file} does not exist, the process will stop.`);
```

And is it wanted to not use `O3rCliError` from `@o3r/schematics` ?

File: packages/@o3r/design/cli/build-design-token.cli.cts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        }
+        return res;
+      })
+      .map(async (file) => ({ file, parsed: await parseDesignTokenFile(file, {specificationContext: { template }}) }))
+  )).reduce<DesignTokenVariableSet>((acc, { file, parsed }) => {
+    parsed.forEach((variable, key) => {
+      if (acc.has(key)) {
+        console.warn(`A duplication of the variable ${key} is found in ${file}`);
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/cli/build-design-token.cli.cts:37]:
```suggestion
        console.warn(`A duplication of the variable ${key} is found in ${file}.`);
```

File: packages/@o3r/design/src/core/design-token/renderers/design-token/design-token-updater.renderers.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    }, { ...tokenObject });
+};
+
+/**
+ * Retrieve a Content Updater function for Metadata generator
+ * @param _options
+ */
+export const getDesignTokenStyleContentUpdater = (_options?: DesignTokenStyleContentUpdaterOptions): DesignContentFileUpdater => {
+  return (variables: string[]) => {
+    const nodes: {[x: string]: any}[] = JSON.parse(`[${variables.join(',')}]`);
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/src/core/design-token/renderers/design-token/design-token-updater.renderers.ts:21]: Why not using `Record<string, any>[]` to be consistent with the line below or the opposite ?

File: .yarnrc.yml [hunk 5]
```diff
@@ -17,12 +21,8 @@ packageExtensions:
...
-    dependencies:
-      "undici": "^5.25.4"
-  "probot@*":
+  probot@*:
     peerDependencies:
-      "@types/express-serve-static-core": "^4.19.5"
+      "@types/express-serve-static-core": ^4.19.5
 
-yarnPath: .yarn/releases/yarn-4.4.0.cjs
+yarnPath: .yarn/releases/yarn-4.4.1.cjs
```
Comment (matthieu-crouzet) [inline .yarnrc.yml:28]: Should it be part of that PR ?

File: packages/@o3r/design/cli/build-design-token.cli.cts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+const args = minimist(process.argv.splice(2));
+
+void (async () => {
+  const renderDesignTokenOptions: DesignTokenRendererOptions = {
+    tokenDefinitionRenderer: getDesignTokenTokenDefinitionRenderer({ keyJoinNumber: args.l || args.level })
+  };
+
+  const output: string = args.o || args.output;
+  const templatePath: string | undefined = args.t || args.template;
+  const template = templatePath ? JSON.parse(await fs.readFile(resolve(process.cwd(), output), { encoding: 'utf8' })) : undefined;
```
Comment (fpaul-1A) [inline packages/@o3r/design/cli/build-design-token.cli.cts:18]:
```suggestion
  const template = templatePath ? JSON.parse(await fs.readFile(resolve(process.cwd(), templatePath), { encoding: 'utf8' })) : undefined;
```
 ?

Comment (kpanot) [reply packages/@o3r/design/cli/build-design-token.cli.cts:18]: Shame on me....

File: packages/@o3r/design/src/core/design-token/renderers/design-token/design-token-updater.renderers.ts (hunk 4 above)
Comment (kpanot) [reply packages/@o3r/design/src/core/design-token/renderers/design-token/design-token-updater.renderers.ts:21]: No reason, I used it in the beginning and forgot to fix this one

File: .yarnrc.yml (hunk 5 above)
Comment (kpanot) [reply .yarnrc.yml:28]: Moved out to #2106

**Review by fpaul-1A — APPROVED** [review]

**Review by matthieu-crouzet — APPROVED** [review]

**Pull request merged**